### PR TITLE
Add instructions for building/testing project in contributing docs

### LINKED ISSUE #1977: CONTRIBUTING.adoc incomplete
It seems there's a gap between CONTRIBUTING.adoc and getting `mvn test` to pass. At a minimum, one also needs to install the Cloud SDK and the cloud datastore emulator. There might well be other setuup steps to perform. We should document these in sufficient detail that a developer can go from `git clone` to `mvn test` passing without having to contact the development team.

### PR DESCRIPTION
**Pull request by dzou**
This adds instructions for building and testing the project in the contributing docs.

Note: I was able to successfully follow the instructions building the project from a fresh docker image.

Fixes #1977.

### CODE REVIEW

File: CONTRIBUTING.adoc [hunk 1]
```diff
@@ -1,13 +1,47 @@
...
+
+We graciously welcome contributions to Spring Cloud GCP.
+
+== Building the Project
+
+The first step to contributing is to fork the repository and clone it onto your machine.
+
+1. Ensure you have Java version 8 or later installed on your machine.
+
+1. Install the https://cloud.google.com/sdk/docs/[Google Cloud SDK].
```
Comment (meltsufin) [inline CONTRIBUTING.adoc]: This should apply only if you want to run integration tests and samples.
It should be possible to run `mvn clean install` without gcloud.
I believe that's how it's built on Jenkins currently.

Comment (dzou) [reply CONTRIBUTING.adoc]: Done; I reorganized the info to express this better.

File: CONTRIBUTING.adoc [hunk 2]
```diff
@@ -1,13 +1,46 @@
-= Code formatting guidelines
+= Contributing to Spring Cloud GCP
+
+We graciously welcome contributions to Spring Cloud GCP.
```
Comment (elefeint) [inline CONTRIBUTING.adoc]: I don't know why, but "graciously" is giving me the eyerolls.

File: CONTRIBUTING.adoc [hunk 3]
```diff
@@ -1,13 +1,46 @@
...
+3. (Optional) Install the https://cloud.google.com/sdk/docs/[Google Cloud SDK].
+The Google Cloud SDK a set of tools that you can use to manage resources and applications hosted on Google Cloud Platform.
++
+For our purposes, it contains the `gcloud` command line tool which allows you to specify a GCP account and project with which you can run our integration tests and sample applications on your machine.
++
+a. Run `gcloud auth application-default login` to log into your https://console.cloud.google.com[Google Cloud Platform account].
++
+b. Run `gcloud config set project [YOUR_PROJECT_ID]` to set the GCP project ID you wish to use.
++
+c. Verify your settings using  `gcloud config list`.
```
Comment (elefeint) [inline CONTRIBUTING.adoc]: There is an extra space after "using".

File: CONTRIBUTING.adoc [hunk 4]
```diff
@@ -1,13 +1,46 @@
...
+1. Ensure you have Java version 8 or later installed on your machine.
+
+2. Try running some tests.
++
+Run `./mvnw clean test` in the root directory of the project to run the tests.
+The `./mvnw` is a self-contained https://maven.apache.org/[Maven] wrapper that allows you to build the project without having Maven installed on your local machine.
++
+You can run the tests of a specific module by using the `-f` flag like this: `./mvnw clean test -f spring-cloud-gcp-pubsub`
+
+3. (Optional) Install the https://cloud.google.com/sdk/docs/[Google Cloud SDK].
```
Comment (meltsufin) [inline CONTRIBUTING.adoc]: For this to be true, we need to guard the emulator-based tests behind a flag.

File: CONTRIBUTING.adoc [hunk 5]
```diff
@@ -20,7 +53,7 @@ Enable the `Eclipse code formatter` by clicking `Use the Eclipse code formatter`
 * IDEA's "Optimize imports on the fly" option turned on interferes with the Eclipse code formatter import optimization.
 Consider disabling the option if optimization does not yield the expected results.
 
-= Contributing a patch
+== Contributing a patch
 
 1. Submit an issue describing your proposed change
```
Comment (elefeint) [inline CONTRIBUTING.adoc]: 2. After working out any ambiguities, create a pull request with the proposed code change.

File: CONTRIBUTING.adoc (hunk 4 above)
Comment (dzou) [reply CONTRIBUTING.adoc]: I ran `./mvnw test` in a clean docker container and it worked for me. It looks like the emulator tests are already disabled by default; I get a message that looks like this: 

```
[INFO] Running org.springframework.cloud.gcp.stream.binder.pubsub.PubSubMessageChannelBinderEmulatorTests
15:47:32.591 [main] WARN  o.s.c.g.s.b.pubsub.PubSubEmulator - PubSubEmulator rule disabled. Please enable with -Dit.pubsub-emulator.
```

File: CONTRIBUTING.adoc (hunk 2 above)
Comment (dzou) [reply CONTRIBUTING.adoc]: Removed :)

File: CONTRIBUTING.adoc (hunk 4 above)
Comment (meltsufin) [reply CONTRIBUTING.adoc]: Hmm, not sure why @artembilan had an issue when running the build locally then.

File: CONTRIBUTING.adoc [hunk 6]
```diff
@@ -53,5 +86,5 @@ To update already-committed files, pass a commit hash or a branch name to use as
 
  $ bash util/update-license-year.sh master
 
-1. Submit a pull request.
-It will be approved and merged by a committer
+1. After working out any ambiguities, create a pull request with the proposed code change.
```
Comment (meltsufin) [inline CONTRIBUTING.adoc]: What's this "ambiguities" about?

File: CONTRIBUTING.adoc (hunk 4 above)
Comment (artembilan) [reply CONTRIBUTING.adoc]: I have problems only when emulator-specific tests are not guarded.
I guess everyone else without Google Cloud SDK will fail the same way...

Comment (dzou) [reply CONTRIBUTING.adoc]: Lol, I think Mike meant to mention Elliotte right? And not Artem?

Comment (meltsufin) [reply CONTRIBUTING.adoc]: Ah, you're right. @elharo

File: CONTRIBUTING.adoc (hunk 6 above)
Comment (dzou) [reply CONTRIBUTING.adoc]: Done, I simplified the sentence.

**Pull request merged**